### ISSUE TITLE
Get p2p_libp2p go tests to pass on Windows

### ISSUE THREAD

**Issue by DavidMinarsch** [fetchai/agents-aea#2279]
**Is your feature request related to a problem? Please describe.**
There are multiple test failures: https://github.com/fetchai/agents-aea/runs/1944083237?check_suite_focus=true

**Describe the solution you'd like**
Investigate and resolve

**Comment by lrahmani** [comment]
We had this issue since we first setup golang checks on ci for Windows and disabled them.
This issue needs more investigation. As of now here is what we know:
-  All tests passes when you run them individually, for e.g.,
   ```powershell
   go test -p 1 -timeout 0 -count 1 -v .\dht\dhtpeer\ -run TestRoutingDHTPeerToDHTPeerIndirectTwoHops
   ```
- When tests are launched for the whole module (i.e., `.\...`)  tests execution slows down quite considerably after the first couple of tests passing
- This slow down makes the remaining tests fail and it never finishes executing all tests
- A similar slow down is perceived on Macos and Ubuntu, but it is very minimal

My current hypothesis is that the issue is related to either or all (i) `go test` tests aggregation (ii) `go test` test cleanup and go runtime runtime scheduling (iii) improper cleanup/termination of goroutines by our go code that is more harmful on Windows specifically

**Comment by DavidMinarsch** [comment]
Looks like recent improvements fixed this #2498
